frontend/src/pages/store/index.svelte

```diff
@@ -95,7 +288,126 @@
 		{:else}
 			<h2>No stores available</h2>
 		{/if}
-	{:catch}
+	{:else}
 		<h2>Error loading stores</h2>
-	{/await}
+	{/if}
 </div>
+
+<input
+	type="checkbox"
+	id="editmodal"
+	class="modal-toggle"
+	bind:checked={modalOpen}
+/>
+<div class="modal">
+	<div class="modal-box">
+		<div class="flex flex-row justify-between items-center mb-4">
+			<h2 class="inline-flex text-2xl text-medium">
+				{modalStore ? `Edit ${modalStore.name}` : 'Create store'}
+			</h2>
+			{#if modalStore && modalStore.owner === userInfo.id}
+				<button
+					class="inline-flex btn btn-circle btn-ghost text-3xl modal-button"
+					on:click={deleteStore}
+				>
+					<span class="icon-delete" />
+				</button>
+			{/if}
+		</div>
+		{#key modalStore}
+			<Form
+				on:submit={doManageStore}
+				on:update={() =>
+					Object.keys(manageFormData).forEach(
+						key => (manageFormData[key] = manageForm[key]),
+					)}
+				bind:this={manageForm}
+				validators={manageValidators}
+			>
+				<Input
+					name="name"
+					label="Name"
+					bind:value={manageFormData.values.name}
+					bind:error={manageFormData.errors.name}
+					on:validate={manageForm.validate}
+				/>
+				<Input
+					name="description"
+					label="Description (optional)"
+					type="textarea"
+					bind:value={manageFormData.values.description}
+					bind:error={manageFormData.errors.description}
+					on:validate={manageForm.validate}
+				/>
+				<ClassroomSearch
+					name="classes"
+					label="Classes who can access this store"
+					bind:value={manageFormData.values.classes}
+					bind:error={manageFormData.errors.classes}
+					on:validate={manageForm.validate}
+					role="teacher"
+					multiselect
+				/>
+				<StudentSearch
+					name="users"
+					label="Additional users who can access this store"
+					bind:value={manageFormData.values.users}
+					bind:error={manageFormData.errors.users}
+					on:validate={manageForm.validate}
+					multiselect
+				/>
+				<StudentSearch
+					name="managers"
+					label="Additional users who can managers this store"
+					bind:value={manageFormData.values.managers}
+					bind:error={manageFormData.errors.managers}
+					on:validate={manageForm.validate}
+					multiselect
+				/>
+				{#if userInfo && userInfo.roles.includes('ADMIN')}
+					<Input
```

Comment:
If `STAFF` can create stores, why can only `ADMIN` users make them public?

Reply:
Most staff will never need to create a public store, so I figured it was best to restrict to admins, especially since a staff could accidentally make it public and not realize.